pages/migration/wsc-53/migration_wsc-53_session.md

```diff
@@ -0,0 +1,180 @@
+	<classname>wcf\system\user\multifactor\FoobarMultifactorMethod</classname>
+</type>
+```
+
+The given classname must implement the [`IMultifactorMethod`](https://github.com/WoltLab/WCF/blob/master/wcfsetup/install/files/lib/system/user/multifactor/IMultifactorMethod.class.php) interface.
+
+As a self-contained example you can find the initial implementation of the email multi-factor method in [WoltLab/WCF#3729](https://github.com/WoltLab/WCF/pull/3729).
+Please check [the version history](https://github.com/WoltLab/WCF/commits/master/wcfsetup/install/files/lib/system/user/multifactor/EmailMultifactorMethod.class.php) of the PHP class to make sure you do not miss important changes that were added later.
+
+{% include callout.html content="Multi-factor authentication is security sensitive.
+Make sure to carefully read the remarks in IMultifactorMethod for possible issues.
+Also make sure to carefully test your implementation against all sorts of incorrect input and consider attack vectors such as race conditions.
+It is strongly recommended to generously check the current state by leveraging assertions and exceptions." type="warning" %}
+
+## Deprecations and Removals
+
+### SessionHandler
+
+Most of the changes with regard to the new session handling happened in `SessionHandler`.
+Most notably `SessionHandler` now is marked `final` to ensure proper encapsulation of data.
+It is not expected that anyone inherited from `SessionHandler`, as it can be considered an implementation detail.
+
+A number of methods in `SessionHandler` are now deprecated and a result in a noop.
+This mostly includes methods that have been used to bootstrap the session, such as `setHasValidCookie()`.
+
+Additionally accessing the following keys on the session is deprecated.
+They directly map to an existing method in another class and any uses can easily be updated:
+- `ipAddress`
+- `userAgent`
+- `requestURI`
+- `requestMethod`
+- `lastActivityTime`
+
+Refer to [the implementation](https://github.com/WoltLab/WCF/blob/439de4963c947c3569a0c584f795245f693155b0/wcfsetup/install/files/lib/system/session/SessionHandler.class.php#L168-L178) for details.
+
+### Cookies
+
+The `_userID`, `_password` and `_cookieHash` cookies will no longer be created and no longer be consumed.
+
+### Virtual Sessions
+
+The virtual session logic existed to support multiple devices per single session in `wcf1_session`.
+Virtual sessions are no longer required with the refactored session handling.
+
+Anything related to virtual sessions has been completely removed as they are considered an implementation detail.
+This removal includes PHP classes and database tables.
+
+### Security Token Constants
+
+The security token constants are deprecated.
+Instead the methods on `SessionHandler` should be used (e.g. `->getSecurityToken()`).
+Within templates it is recommended to use the `{csrfToken}` tag in place of `{@SECURITY_TOKEN_INPUT_TAG}`.
```

Comment:
```suggestion
Within templates, it is recommended to use the `{csrfToken}` tag in place of `{@SECURITY_TOKEN_INPUT_TAG}`.
```

Is it only recommended or should it be done. I would word it more strongly in the second case.